ui/app/components/app/permission-page-container/permission-page-container-content/permission-page-container-content.component.js

```diff
@@ -99,26 +102,93 @@ export default class PermissionPageContainerContent extends PureComponent {
     )
   }
 
-  render () {
-    const { domainMetadata, redirect, permissionRejected, selectedIdentities } = this.props
+  getAccountDescriptor (identity) {
+    return `${identity.label} (...${identity.address.slice(identity.address.length - 4)})`
+  }
+
+  renderAccountTooltip (textContent) {
+    const { selectedIdentities } = this.props
+    const { t } = this.context
+
+    return (
+      <Tooltip
+        key="all-account-connect-tooltip"
+        position="bottom"
+        wrapperClassName="permission-approval-container__bold-title-elements"
+        html={(
+          <div style={{ display: 'flex', flexDirection: 'column' }}>
+            { selectedIdentities.slice(0, 6).map((identity, index) => {
+              return (
+                <div key={ `tooltip-identity-${index}` }>
+                  { this.getAccountDescriptor(identity) }
+                </div>
+              )
+            }) }
+            { selectedIdentities.length > 6
+              ? t('plusXMore', [ selectedIdentities.length - 6 ])
+              : null
+            }
+          </div>
+        )}
+      >{ textContent }
+      </Tooltip>
+    )
+  }
+
+  renderTitleSubstituteText (name, key = 'title-substitute-text') {
```

Comment:
Nit: I don't follow what this name was supposed to mean, nor why the `key` has this default value. I'm guessing this was intended to be reused but never was?

It would make more sense to me to get rid of the `key` parameter and make this function specific to this scenario (e.g. `renderSingleAccountTitle`)